dot/network/status.go

```diff
@@ -121,8 +122,10 @@ func (status *status) validMessage(msg *StatusMessage) bool {
 		return false
 	}
 
+	log.Debug("[network]", "peer's genesis hash", msg.GenesisHash)
+
 	switch {
-	case msg.GenesisHash != status.hostMessage.GenesisHash:
+	case !bytes.Equal(msg.GenesisHash[:], status.hostMessage.GenesisHash[:]):
```

Comment:
was this the solution to the genesis hash mismatch error?

Reply:
I don't think so, it seems to have improved but I did run into it once still, after running nodes for a long time then stopping/restarting